### PR TITLE
Fix `conversation-filters` showing multiple checked marks

### LINKED ISSUE #4589: `conversation-filters` shows the wrong "current filter" checkbox
### Please ensure:

- [X] The bug is caused by Refined GitHub. It doesn't happen if I disable the extension.

### Describe the problem and how to replicate it

"Everything mentioning you" is cloned as a whole and the checkbox is preserved, but it shouldn't be

### Screenshots/video/gif showing the issue, if it’s visual

Screen Shot

### Console errors, if any

_No response_

### Example URL

```js
// https://github.com/sindresorhus/refined-github/issues?q=is%3Aopen+mentions%3A%40me
```

### Browser(s) used

safari

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fixes #4589 

## Test URLs
```js
// https://github.com/refined-github/refined-github/issues?q=is%3Aopen+mentions%3A%40me
```

## Screenshot

image

### CODE REVIEW

File: source/features/conversation-filters.tsx [hunk 1]
```diff
@@ -12,12 +12,14 @@ function init(): void {
 	commentsLink.lastChild!.textContent = 'Everything commented by you';
 	commentsLink.removeAttribute('target');
 	new SearchQuery(commentsLink).set('is:open commenter:@me');
+	commentsLink.setAttribute('aria-checked', String(commentsLink.href === location.href)); // #4589
```
Comment (cheap-glitch) [inline source/features/conversation-filters.tsx:15]: This allow marking the filter as selected, just like the native ones.

image

It follows the behavior of the native filters, that are only shown selected when the current search query matches exactly with the one they represent.

File: source/features/conversation-filters.tsx [hunk 2]
```diff
@@ -12,12 +12,14 @@ function init(): void {
...
 
 	sourceItem.after(commentsLink);
 
 	// Add "Everything you subscribed to" link
 	const subscriptionsLink = commentsLink.cloneNode(true);
 	subscriptionsLink.lastChild!.textContent = 'Everything you subscribed to';
+	subscriptionsLink.setAttribute('aria-checked', 'false'); // #4589
```
Comment (cheap-glitch) [inline source/features/conversation-filters.tsx:22]: We never want this to be selected since it's just a link to another page.

**Comment by cheap-glitch** [comment]
I've noticed an inconsistent behavior due to `sort-conversations-by-update-time`: when first selecting a filter (native or not), the results are going to be ordered by update time, which means the query will not be exactly the one of the filter and thus it will not show up as selected.

What's even more confusing, when selecting the filter again, this time it will work as expected because the sorting feature wasn't reapplied due to the AJAX navigation.

https://user-images.githubusercontent.com/46634000/141992334-904349e2-84a3-40b8-82a7-25f844d53bb0.mp4

So imo `sort-conversations-by-update-time` should not apply to conversation filters at all.

**Comment by kidonng** [comment]
> I've noticed an inconsistent behavior due to `sort-conversations-by-update-time`: when first selecting a filter (native or not), the results are going to be ordered by update time, which means the query will not be exactly the one of the filter and thus it will not show up as selected.

A similar bug is reported in #4891. While I think sorting by update time is helpful, it should not _override_ what the filters should do.

**Comment by fregante** [comment]
> While I think sorting by update time is helpful, it should not _override_ what the filters should do.

It should. That feature is meant to change the default of the website everywhere, so unless you intend to change the sorting, then _every link_ should be affected.

I suppose that the drawback of this are widgets that have a strict "current" implementation. For example on the global PR list page `(/pulls)` the filters continue working even with the extra `sort:`

**Comment by fregante** [comment]
> the sorting feature wasn't reapplied due to the AJAX navigation

That's a bug (of the sorting feature)



> I've noticed an inconsistent behavior due to [`sort-conversations-by-update-time`](/refined-github/refined-github/blob/main/source/features/sort-conversations-by-update-time.tsx)

I suppose that that feature would have to also deal with the "current" implementation 😬 which I wouldn't be a fan of… so hopefully there's an elegant solution.

**Comment by cheap-glitch** [comment]
Opened #5090 to fix the issues with `sort-conversations-by-update-time`.

**Review by fregante — APPROVED** [review]
Untested but TMNT

**Pull request merged**